doc/application/index.rst

```diff
@@ -689,27 +689,43 @@ the Zephyr tree, for example:
 
 
 
-The paths to any Kconfig files inside the structure needs to prefixed with
-$(SOC_DIR) to make Kconfig aware of the location of the Kconfig files related to
-the custom SOC.
+The Zephyr :file:`soc/Kconfig` will create the top-level
+``SoC/CPU/Configuration Selection`` menu in Kconfig.
 
-In the ``soc`` directory you will need a top-level Kconfig file pointing to the
-custom SOC definitions:
+By providing additional ``SOC_ROOT`` then it is possible to load out-of-tree
+SoCs into this menu.
 
+Following the structure above, then the following files can be added to load
+more SoCs into the menu.
 
 .. code-block:: none
 
-   choice
-   	prompt "SoC/CPU/Configuration selection"
+        soc
+        └── arm
+            └── st_stm32
+                    ├── Kconfig
+                    ├── Kconfig.soc
+                    └── Kconfig.defconfig
```

Comment:
Can you please emulate the board porting guide by describing these files' contents in detail?

https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#write-kconfig-files

The differences between these files is not clear and IMO should be.

Perhaps the whole thing should be moved to a new "SoC porting guide" and just referenced from here.

Reply:
I agree, but this has been the case always and is not related to the SOC_ROOT changes.

If acceptable to you, then I would like to do a followup description PR on how to describe out-of-tree soc.
The reason for taking it separately is mostly to get this PR in as fast as possible, so that people become aware of the SOC_ROOT changes.

A dedicated SOC porting guide can then follow, as that is something new.

Reply:
> I agree, but this has been the case always and is not related to the SOC_ROOT changes.

Are you sure? Didn't this new change add new files?

Reply:
No, those file where indirectly mentioned as `source` / `osource` in old example:
https://docs.zephyrproject.org/latest/application/index.html#soc-definitions
```
choice
     prompt "SoC/CPU/Configuration selection"

source "$(SOC_DIR)/$(ARCH)/*/Kconfig.soc"

endchoice

menu "Hardware Configuration"
osource "$(SOC_DIR)/$(ARCH)/*/Kconfig"

endmenu
```
as you see, it was presented how to create the menu and then source those files, but without describing their content.

Reply:
> If acceptable to you, then I would like to do a followup description PR on how to describe out-of-tree soc.

Sounds good, I made https://github.com/zephyrproject-rtos/zephyr/issues/27439 so we don't forget.